asteroid/data/avspeech_dataset.py

```diff
@@ -1,25 +1,186 @@
+
+    def augment_audio(self, augmenter: Callable, *args, **kwargs):
+        '''
+            Change the audio via the augmenter method.
+        '''
+        self.audio = augmenter(self.audio, *args, **kwargs)
+
+    def augment_video(self, augmenter: Callable, *args, **kwargs):
+        '''
+            Change the video via the augmenter method.
+        '''
+        self.video = augmenter(self.video, *args, **kwargs)
+
+    def _convert_video(self):
+        if self.embed_saved:
+            return
+        self.buffer_video = get_frames(self.video)
+
+    def _check_video_embed(self, embed_ext=".npy"):
+        video_name_stem = self.video_path.stem
+
+        embed_dir = self.embed_dir
+        if not embed_dir.is_dir():
+            embed_dir = Path(*embed_dir.parts[2:])
+
+        self.embed_path = Path(embed_dir, video_name_stem + f"_part{self.video_start_length}" + embed_ext)
+        if self.embed_path.is_file():
+            self.embed_saved = True
+            self.embed = np.load(self.embed_path.as_posix())
+
+    def embed_is_saved(self):
+        return self.embed_saved
+
+    def get_embed(self):
+        return self.embed
+
+    def get_video(self):
+        #retrieve slice of video, if video > 75 frames
+        buffer_video = self.buffer_video[self.video_start_length*75: (self.video_start_length+1)*75]
+        return buffer_video
+
+    def get_audio(self):
+        return self.audio
+
+    def get_spec(self):
+            return np.load(self.spec_path)
+
+    def is_spec(self):
+        return self._is_spec
+
+    @staticmethod
+    def load_audio(audio_path: str, sr=16_000, load_spec=False):
+        audio_path = Path(audio_path)
+        spec_exists = False
+        spec_path = Path(*audio_path.parts[:-2], "spec", audio_path.stem + ".npy")
+        if load_spec and spec_path.is_file():
+            spec_exists = True
+            audio = np.load(spec_path)
+        else:
+            audio = librosa.load(str(audio_path), sr=sr)[0]
+        return audio, spec_exists, spec_path
+
 
 class AVSpeechDataset(data.Dataset):
 
     def __init__(self, input_df_path: Path,
-                signal_constructor, convert_to_spectrogram,
-                input_audio_size=2):
+                 embed_dir: Path,
+                 input_audio_size=2):
         """
 
             Args:
                 input_df_path: path for combination dataset
                 input_audio_size: total audio/video inputs
         """
         self.input_audio_size = input_audio_size
+        self.embed_dir = embed_dir
         self.input_df = pd.read_csv(input_df_path.as_posix())
-        self.signal_constructor = signal_constructor
-        self.convert_to_spectrogram = convert_to_spectrogram
+
+    @staticmethod
+    def encode(x: np.ndarray, p=0.3, stft_encoder=None):
+        if stft_encoder is None:
+            stft_encoder = Encoder(STFTFB(n_filters=512, kernel_size=400,
+                                          stride=160))
```

Comment:
This will create an encoder everytime, it would be better to define these in `__init__`. 
In this case, you can either use self.stft_encoder, or keep themethod static and pass the encoder but why would you like to keep this static?

Reply:
We would still need both of these methods during inference. So, we will have to create the `AVSpeechDataset` object during inference as well.

Reply:
Maybe, this is related to the general comment I made. We would like a class that handles feature computation and audio-video alignement (which is configurable as well). And this class would be used in the `Dataset` object, but this would also be the class that we could use at inference to feed the network with data. Would this make sense?